### DISCUSSION TITLE
Re Macports, sandbox and 10.6 PPC

### DISCUSSION THREAD

**Discussion by barracuda156** [barracuda156/10A190#1] [General]
@kencu @evanmiller I don’t know where to ask this otherwise, so making a topic here. _Any advice is greatly appreciated._

I have recently installed 10A96 build, available now here: https://macintoshgarden.org/sites/macintoshgarden.org/files/apps/10A096_Clean.dmg

Turned out that the @kencu steps to set up Macports which worked for 10A190 perfectly _fail for 10A96_ on sandbox part (relevant setting has been changed in portsandbox.tcl to >10).

I do not expect anyone to take pain installing 10A96, please just look at the error. It is 100% consistent (does not depend on a port): `sandbox_init: Could not set default launchd policy, err=1100`.
Example:

```
--->  Checksumming neofetch-7.1.0.tar.gz
--->  Extracting neofetch
--->  Extracting neofetch-7.1.0.tar.gz
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_neofetch/neofetch/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/neofetch/neofetch-7.1.0.tar.gz' | /usr/bin/gnutar --no-same-owner -xf -
sandbox_init: Could not set default launchd policy, err=1100
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_sysutils_neofetch/neofetch/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/neofetch/neofetch-7.1.0.tar.gz' | /usr/bin/gnutar --no-same-owner -xf -
Exit code: 2
Error: Failed to extract neofetch: command execution failed
```

That is, no ports can be installed, they fail on extraction step. To re-iterate, I have followed the procedure which works for 10A190 precisely. I have set up Macports for 10.6 PPC multiple times, nothing was missed. Error was also reproduced by another person on a different machine with 10A96. There is some weird thing with OS itself, apparently, but maybe it is still fixable?

**Comment by kencu** [comment]
I don't see any changes in portsandbox.tcl

https://github.com/macports/macports-base/commits/master/src/port1.0/portsandbox.tcl

or in portextract.tcl

https://github.com/macports/macports-base/commits/master/src/port1.0/portextract.tcl

that look like they could cause this error.


I have sometimes had issues when there is no macports user and group, or when the macports home directory has different ownership than the user macports is running as. That is worth a quick check perhaps.

**Comment by kencu** [comment]
@jmroot might have time to give you a pointer.

**Comment by barracuda156** [comment]
@kencu Thank you!

Prior to writing a question here I have examined the whole Macports libexec contents with multi-file search looking for anything sandbox-related that makes difference between Darwin versions or other suggestive Darwin9/10-specific settings and failed to find any. I also compared sandbox.h between 10A96 and 10A190, they are identical. So I was out of ideas.

**Comment by barracuda156** [comment]
@jmroot If you may give an advice of the matter, will be awesome!
